src/components/Courses/CoursesCollection/index.tsx

```diff
@@ -146,233 +205,292 @@ const CoursesList = styled.section`
 const CoursesCollection: React.FC = () => {
   const { params, setParams, courses } = useContext(CoursesContext);
   const { categoryTree, uniqueTags } = useContext(EscolaLMSContext);
-  const [selectedMainCategory, setSelectedMainCategory] =
-    useState<API.Category | null>(null);
-  const [selectedSubCategory, setSelectedSubCategory] =
-    useState<API.Category | null>(null);
-  const [selectedMobileCategories, setselectedMobileCategories] = useState<
-    number[]
-  >([]);
+  const [parsedParams, setParsedParams] = useState<
+    API.CourseParams | undefined
+  >();
+  const [selectedCategories, setSelectedCategories] = useState<number[]>([]);
+  const [typeSelectedValue, setTypeSelectedValue] = useState<
+    string | undefined
+  >(undefined);
+  const [tagSelectValue, setTagSelectValue] = useState<string | undefined>(
+    undefined
+  );
```

Comment:
if you're using this to call API it should be one state 

```suggestion
  const [filterState, setFilterState] = useState<{categories: number[], type?: string, tag?: string>([]);
```

then you update something link 

```ts
type updateParamType  = { key: "type" | "tag", value: string | undefined } | { key:"categories", value: number[]} ;
const updateState = useCallback((updateObj:updateParamType) => setFilterState(prevState => {...prevState, [key]: value}), []);

updateState({key:"categories", value: [1,2,3]});
```

Reply:
then the rest of the code should be much less complex 

also remeber that the current state should be in the URL so 
- when you click on change filter url change 
- when you go on this page with param like ?categories=1,2,3 it should fetch them 
- when you press prev/next button in the browser it should change 

the actual call should be made **only** on 
- change and init of URL params 
- never on pressing button (pressing buttons should change the url) 
- query params is good entry example https://v5.reactrouter.com/web/example/query-parameters